server/gokbg3/grails-app/domain/org/gokb/cred/Source.groovy

```diff
@@ -142,4 +144,37 @@ class Source extends KBComponent {
       }
     }
   }
+
+  public boolean needsUpdate() {
+    Source src = this
+    if (src.lastRun == null) {
+      return true;
+    }
+    if (src.frequency != null) {
+      Pattern pattern = Pattern.compile("\\s*(\\d*)\\s*([a-zA-Z]*)\\s*");
+      Matcher matcher = pattern.matcher(src.frequency);
+      int days
+      if (matcher.find()) {
+        def length = [
+          D: 1,
```

Comment:
This is OK for now, but would need to be refined. Chosen the option "M1" for example, the user would expect the update to happen at the very same day of each month.

Reply:
M1 won't pass the parser, as it expects the number *before* the letter. Maybe the user expects the update at the same day-of-month, though I highly doubt that he or she notices the slight shift of 2 days max, within one month.